.github/workflows/tests.yml

```diff
@@ -24,15 +24,29 @@ jobs:
       fail-fast: false
       matrix:
         os: ["windows-latest", "ubuntu-latest", "macos-latest"]
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        environment: ["3.8", "3.9", "3.10", "3.11"]
+        extra: [null]
         exclude:
-            - os: "macos-latest"
-              python-version: "3.8"
-            - os: "macos-latest"
-              python-version: "3.10"
+          - os: "macos-latest"
+            environment: "3.8"
+          - os: "macos-latest"
+            environment: "3.10"
+        include:
+          # Minimum dependencies
+          - os: "ubuntu-latest"
+            environment: "mindeps-array"
+          - os: "ubuntu-latest"
+            environment: "mindeps-dataframe"
+          - os: "ubuntu-latest"
+            environment: "mindeps-distributed"
+          - os: "ubuntu-latest"
+            environment: "mindeps-non-optional"
+          # Pyarrow strings turned on
+          - os: "ubuntu-latest"
+            environment: "3.10"
```

Comment:
Why not the latest (3.11)?

Reply:
3.11 has a few additional dependencies commented out (like `numba` and `pyspark`) because they don't have Python 3.11 support yet. I went with 3.10 here to make sure we have the broadest test coverage